File: backend/customer-service/src/main/java/com/crm/customer/model/Customer.java
package com.crm.customer.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "crm_customers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(name = "contact_person", length = 120)
    private String contactPerson;

    @Column(nullable = false, length = 150)
    private String email;

    @Column(length = 50)
    private String phone;

    @Column(length = 150)
    private String company;

    @Column(length = 200)
    private String website;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private Industry industry = Industry.TECHNOLOGY;

    @Enumerated(EnumType.STRING)
    @Column(name = "customer_tier", nullable = false, length = 50)
    @Builder.Default
    private CustomerTier customerTier = CustomerTier.TIER_3_SMB;

    @Enumerated(EnumType.STRING)
    @Column(name = "customer_status", nullable = false, length = 50)
    @Builder.Default
    private CustomerStatus customerStatus = CustomerStatus.ACTIVE;

    @Column(name = "annual_revenue", precision = 15, scale = 2)
    @Builder.Default
    private BigDecimal annualRevenue = BigDecimal.ZERO;

    @Column(name = "billing_address", length = 300)
    private String billingAddress;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "assigned_account_manager_id")
    private Long assignedAccountManagerId;

    @Column(name = "assigned_account_manager_name", length = 120)
    private String assignedAccountManagerName;

    @Column(name = "converted_from_lead_id")
    private Long convertedFromLeadId;

    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Boolean isDeleted = false;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @Column(name = "deleted_by_user_id")
    private Long deletedByUserId;

    @Column(name = "deleted_by_user_name", length = 120)
    private String deletedByUserName;

    @Column(name = "deleted_by_role", length = 50)
    private String deletedByRole;

    @Column(name = "delete_requested", nullable = false)
    @Builder.Default
    private Boolean deleteRequested = false;

    @Column(name = "delete_request_reason", length = 300)
    private String deleteRequestReason;

    @Column(name = "created_by_user_id")
    private Long createdByUserId;

    @Column(name = "created_by_user_name", length = 120)
    private String createdByUserName;

    @Column(name = "created_by_role", length = 50)
    private String createdByRole;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/customer.ts
export type CustomerStatus = 'ACTIVE' | 'ONBOARDING' | 'AT_RISK' | 'CHURNED' | 'INACTIVE';

export type CustomerTier = 'TIER_1_ENTERPRISE' | 'TIER_2_MID_MARKET' | 'TIER_3_SMB' | 'STRATEGIC';

export type Industry = 
  | 'TECHNOLOGY'
  | 'FINANCE'
  | 'HEALTHCARE'
  | 'MANUFACTURING'
  | 'RETAIL'
  | 'EDUCATION'
  | 'SERVICES'
  | 'OTHER';

export interface CustomerProduct {
  id: number;
  customerId: number;
  productId: number;
  productName: string;
  dealId?: number;
  quantity: number;
  unitPrice: number;
  totalAmount: number;
  status: string; // ACTIVE, EXPIRED, CANCELLED
  purchaseDate: string;
  startDate?: string;
  expiryDate?: string;
  billingFrequency?: string;
}

export interface Customer {
  id: number;
  name: string;
  contactPerson?: string;
  email: string;
  phone?: string;
  company?: string;
  website?: string;
  industry: Industry;
  industryDisplayName: string;
  customerTier: CustomerTier;
  tierDisplayName: string;
  customerStatus: CustomerStatus;
  statusDisplayName: string;
  annualRevenue: number;
  billingAddress?: string;
  notes?: string;
  assignedAccountManagerId?: number;
  assignedAccountManagerName?: string;
  convertedFromLeadId?: number;
  isDeleted?: boolean;
  deletedAt?: string;
  deletedByUserId?: number;
  deletedByUserName?: string;
  deletedByRole?: string;
  deleteRequested?: boolean;
  deleteRequestReason?: string;
  createdByUserId?: number;
  createdByUserName?: string;
  createdByRole?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateCustomerRequest {
  name: string;
  contactPerson?: string;
  email: string;
  phone?: string;
  company?: string;
  website?: string;
  industry?: Industry;
  customerTier?: CustomerTier;
  customerStatus?: CustomerStatus;
  annualRevenue?: number;
  billingAddress?: string;
  notes?: string;
  assignedAccountManagerId?: number;
  assignedAccountManagerName?: string;
  convertedFromLeadId?: number;
  createdByUserId?: number;
  createdByUserName?: string;
  createdByRole?: string;
}

export interface UpdateCustomerRequest {
  name: string;
  contactPerson?: string;
  email: string;
  phone?: string;
  company?: string;
  website?: string;
  industry?: Industry;
  customerTier?: CustomerTier;
  customerStatus?: CustomerStatus;
  annualRevenue?: number;
  billingAddress?: string;
  notes?: string;
  assignedAccountManagerId?: number;
  assignedAccountManagerName?: string;
}

export interface CustomerStats {
  totalCustomers: number;
  activeCustomers: number;
  onboardingCustomers: number;
  atRiskCustomers: number;
  churnedCustomers: number;
  trashCustomersCount?: number;
  totalAnnualRevenue: number;
  activeAnnualRevenue: number;
  retentionRate: number;
  customersByTier: Record<string, number>;
  customersByIndustry: Record<string, number>;
  customersByStatus: Record<string, number>;
}
